cmd/grafiti/main.go

```diff
@@ -89,23 +168,26 @@ func initConfig() {
 	viper.SetDefault("maxNumRequestRetries", 8)
 
 	// If a config file is found, read in its data
-	if err := viper.ReadInConfig(); err == nil {
-		logrus.Info("Using config file: ", viper.ConfigFileUsed())
+	if err := viper.ReadInConfig(); err != nil {
+		logger.Errorln("read config file:", err)
+	} else {
+		logger.Infoln("Using config file:", viper.ConfigFileUsed())
 		return
 	}
 
 	// No config file found so configure grafiti with a dummy config file (Reader)
 	// and environment variables
 	viper.SetConfigType("toml")
-	if err := viper.ReadConfig(bytes.NewBuffer([]byte(""))); err == nil {
-		logrus.Info("Using environment variables to configure grafiti.")
+	if err := viper.ReadConfig(bytes.NewBuffer([]byte(""))); err != nil {
+		logger.Errorln("read dummy config file:", err)
+	} else {
+		logger.Info("Using environment variables to configure grafiti.")
 		return
 	}
-
 }
 
 func main() {
 	if err := RootCmd.Execute(); err != nil {
```

Comment:
Instead of `exitWithSuccess` and `exitError`, why don't you use define an Error type for each of the return codes you want, define the command with `RunE`, and then exit with an errorcode defined on the Error type here? Anywhere you want to exitWithSuccess you can return `nil`, everywhere you want to error with a specific code you return that Error type.